pallets/parachain-staking/src/auto_compound.rs

```diff
@@ -63,6 +63,10 @@ where
 		Self(sorted_delegations)
 	}
 
+	pub fn get_auto_compounding_delegation_count(candidate: &T::AccountId) -> usize {
+		<AutoCompoundingDelegationsStorage<T>>::decode_len(candidate).unwrap_or_default()
```

Comment:
One other thing to be careful about is `unwrap_or_default`. I think the only time we should hit this is when the storage item doesn't exist, so treating that as `0` (`default`) is probably ok.

It can also be hit if there's a decode failure, but I suspect that can only happen if the storage item is corrupt...

Reply:
Agree. Other than that the `unwrap_or_default`s I used are compatible with the implementation before.